core/src/widgets/audio_widgets/normalized_map.rs

```diff
@@ -0,0 +1,424 @@
+pub trait NormalizedMap: 'static {
+    fn normalized_to_display(&self, normalized: f32) -> String;
+
+    fn snap(&self, normalized: f32) -> f32 {
+        normalized
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum DisplayDecimals {
+    Zero,
+    One,
+    Two,
+    Three,
+    Four,
+    Five
+}
+
+impl DisplayDecimals {
+    pub fn display_value(&self, value: f32) -> String {
+        match self {
+            DisplayDecimals::Zero => format!("{:.0}", value),
+            DisplayDecimals::One => format!("{:.1}", value),
+            DisplayDecimals::Two => format!("{:.2}", value),
+            DisplayDecimals::Three => format!("{:.3}", value),
+            DisplayDecimals::Four => format!("{:.4}", value),
+            DisplayDecimals::Five => format!("{:.5}", value),
+        }
+    }
+}
+
+impl Default for DisplayDecimals {
+    fn default() -> Self {
+        DisplayDecimals::One
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Gradient {
```

Comment:
There might be a name clash here with gradient paint on the render side. Maybe raname this to something like Scale or ValueScale? SliderScale?